src/treemodel/selection.js

```diff
@@ -83,11 +86,26 @@ export default class Selection {
 	}
 
 	/**
+	 * Flag indicating whether the selection has any range in it.
+	 *
+	 * @readonly
+	 * @type {Boolean}
+	 */
+	get hasAnyRange() {
+		return this._ranges.length > 0;
+	}
+
+	/**
 	 * Returns whether the selection is collapsed. Selection is collapsed when all it's ranges are collapsed.
+	 * If selection has no ranges, returns null instead.
 	 *
 	 * @type {Boolean}
 	 */
 	get isCollapsed() {
+		if ( !this.hasAnyRange ) {
+			return null;
```

Comment:
It should not be null according to the docs.

I would return `true` if the selection is empty. This is what the native selection does in Firefox and it was my first choice as well.